src/Relations/BelongsToMany.php

```diff
@@ -190,10 +190,10 @@ public function attach($id, array $attributes = [], $touch = true)
 
             $query = $this->newRelatedQuery();
 
-            $query->whereIn($this->related->getKeyName(), (array) $id);
+            $query->whereIn($this->relatedKey, (array) $id);
```

Comment:
As I said in the previous comment, `relatedKey` is `$this->related->getKeyName()` by default but respects our override.